src/util-inl.h

```diff
@@ -445,6 +445,33 @@ v8::MaybeLocal<v8::Value> ToV8Value(v8::Local<v8::Context> context,
   return v8::Number::New(isolate, static_cast<double>(number));
 }
 
+template <typename T>
+v8::Local<v8::Array> ToV8ValuePrimitiveArray(v8::Local<v8::Context> context,
+                                             const std::vector<T>& vec,
+                                             v8::Isolate* isolate) {
+  if (isolate == nullptr) isolate = context->GetIsolate();
+  v8::EscapableHandleScope handle_scope(isolate);
+  v8::Local<v8::Array> arr = v8::Array::New(isolate, vec.size());
+
+  for (size_t i = 0; i < vec.size(); i++) {
+    v8::Local<v8::Value> element;
+
+    if constexpr (std::is_same_v<T, bool>) {
+      element = v8::Boolean::New(isolate, vec[i]);
+    } else if constexpr (std::is_same_v<T, int32_t>) {
+      element = v8::Integer::New(isolate, vec[i]);
+    } else if constexpr (std::is_same_v<T, uint32_t>) {
+      element = v8::Integer::NewFromUnsigned(isolate, vec[i]);
+    } else if constexpr (std::is_same_v<T, double>) {
+      element = v8::Number::New(isolate, vec[i]);
+    } else {
+      element = ToV8Value(context, vec[i], isolate).ToLocalChecked();
+    }
+    arr->Set(context, i, element).Check();
+  }
+  return handle_scope.Escape(arr);
+}
```

Comment:
While I'm fine with adding a new `ToV8Value` variant in general, I'd prefer to do so only if there are places it is used. Are there existing call sites that would use this?

Reply:
This won't be the same exact replacement, since if one of the ToV8Value calls fail, it returns and stop execution. But in your example and code, it throws.

Reply:
Creating primitives and arrays of primitives do not actually throw - both `v8::Array::New` and `v8::Integer::New` have non-throwing variants that just copy things, for example. So it's just a variant of `ToV8Value` for converting values that would not have thrown in the first place, and would be more efficient than using the old one because we were dealing with non-existent empty maybes just because the old API was too one-size-fits-all.

Reply:
@Aditi-1400 ... my point on the usage is that we shouldn't land *this* PR without those usages in place. If this can be used to improve various callsites, then those should be updated here also.

Reply:
I couldn't find any other uses of this, most of `ToV8Value`s were being used for strings, string vectors etc. I'll keep looking though